syntax = "proto3";

message Client {
  string name = 1;
}

message Empty {

}

message ChatMessage {
  string sender_name = 1;
  string text = 2;
  int32 timestamp = 3;
}

service ChatService {
  rpc SendMessage(ChatMessage) returns (Empty);
  rpc ReceiveMessage(Empty) returns (stream ChatMessage);
  rpc RegisterClient(Client) returns (Empty);
}
